IDinsight/aaq-core
# Basic framework

## Overview

Sets up the basic framework for the app

## Details

Started setting up backend. At the moment, it has three endpoints: `embeddings-search`, `feedback`, and `healthcheck`
Also setup alembic for migrations.

## Testing it out

1. Create a docker container running postgres
```
docker run --name testdb -e POSTGRES_PASSWORD=postgres -p 5432:5432 -d postgres
```
Note that this will not persist data/schema.

2. Create conda environment and install packages from `requirements.txt` (and `requirements-dev.txt`) 
3. From `aaq-core/core_backend`, run:

```
 ./startup.sh
```

4. Send message to `embedding-search` endpoint

```
curl -X POST -S -d '{"query_text" : "hello" }' -H 'Content-Type: application/json' localhost:8000/embeddings-search
```
You should get "Added : #<ID>" as response

5. Send message to `feedback` endpoint

```
curl -X POST -S -d '{"query_id":"1","feedback_secret_key":"abc", "feedback_text":"this is feedback" }' -H 'Content-Type: application/json' localhost:8000/feedback
```
You should get "Added : #<FEEDBACK_ID>" as response
